src/pseudopeople/interface.py

```diff
@@ -50,61 +50,83 @@ def _generate_form(
-        if not isinstance(data, pd.DataFrame):
-            raise TypeError(
-                f"File located at {data_path} must contain a pandas DataFrame. "
-                "Please provide the path to the unmodified root data directory."
-            )
-
-        # Coerce dtypes prior to noising to catch issues early as well as
-        # get most columns away from dtype 'category' and into 'object' (strings)
-        for col in columns_to_keep:
-            if col.dtype_name != data[col.name].dtype.name:
-                data[col.name] = data[col.name].astype(col.dtype_name)
+        data = _load_data_from_path(data_path, year_filter)
 
+        data = _coerce_dtypes(data, form)
+        data = _reformat_dates_for_noising(data)
         noised_data = noise_form(form, data, configuration_tree, seed)
-        noised_data = _extract_columns(columns_to_keep, noised_data)
+        noised_data = _extract_columns(form.columns, noised_data)
         noised_form.append(noised_data)
 
     noised_form = pd.concat(noised_form, ignore_index=True)
 
     # Known pandas bug: pd.concat does not preserve category dtypes so we coerce
     # again after concat (https://github.com/pandas-dev/pandas/issues/51362)
-    for col in columns_to_keep:
-        if col.dtype_name != noised_form[col.name].dtype.name:
-            noised_form[col.name] = noised_form[col.name].astype(col.dtype_name)
+    noised_form = _coerce_dtypes(noised_form, form)
 
     return noised_form
 
 
+def _coerce_dtypes(data: pd.DataFrame, form: Form):
+    # Coerce dtypes prior to noising to catch issues early as well as
+    # get most columns away from dtype 'category' and into 'object' (strings)
+    for col in form.columns:
+        if col.dtype_name != data[col.name].dtype.name:
+            data[col.name] = data[col.name].astype(col.dtype_name)
+    return data
+
+
+def _load_data_from_path(data_path: Path, year_filter: Dict):
+    """Load data from a data file given a data_path and a year_filter."""
+    if data_path.suffix == ".hdf":
+        with pd.HDFStore(str(data_path), mode="r") as hdf_store:
+            data = hdf_store.select("data", where=year_filter["hdf"])
+    elif data_path.suffix == ".parquet":
+        data = pq.read_table(data_path, filters=year_filter["parquet"]).to_pandas()
+    else:
+        raise ValueError(
+            "Source path must either be a .hdf or a .parquet file. Provided "
+            f"{data_path.suffix}"
+        )
+    if not isinstance(data, pd.DataFrame):
+        raise TypeError(
+            f"File located at {data_path} must contain a pandas DataFrame. "
+            "Please provide the path to the unmodified root data directory."
+        )
+    return data
+
+
+def _reformat_dates_for_noising(data: pd.DataFrame):
+    """Formats SSA event_date and dates of birth, so they can be noised."""
+    if COLUMNS.ssa_event_date.name in data.columns:
+        # event_date -> YYYYMMDD
+        data[COLUMNS.ssa_event_date.name] = pd.to_datetime(
```

Comment:
I think pd.to_datetime by default tries to be smart about the dates...it might be worth using `pd.to_datetime(..., format=xxx)` to ensure months/days are not swapped on accident.

Reply:
Are these columns not datetime dtyped coming out of post-processing? If they are you shouldn't need the `pd.to_datetime` call at all.

Reply:
They come through as objects. I need the to_datetime to be able to use the dt accessor.

Reply:
It's coming in as strings because we are coercing dtypes prior to this formatting. I think if you just move this formatting to before the _coerce_dtype method it should work as Rajan says

Reply:
In fact, moving all of this into the _coerce_dtypes might make sense?

Reply:
I missed this before merging, but I can revisit with another PR.

Reply:
We don't want to move into _coerce_dtypes because we only want the date formatting to happen before noising.

Reply:
PR for this https://github.com/ihmeuw/pseudopeople/pull/64